google-cloud-storage/test/google/cloud/storage/project_test.rb

```diff
@@ -560,6 +560,43 @@ def creds.is_a? target
     _(bucket.rpo).must_equal "ASYNC_TURBO"
   end
 
+  it "created a bucket with HNS enabled" do
+    mock = Minitest::Mock.new
+    created_bucket = create_bucket_gapi bucket_name
+    created_bucket.hierarchical_namespace = hierarchical_namespace_object
+    resp_bucket = bucket_with_location created_bucket
+
+    mock.expect :insert_bucket, resp_bucket, [project, created_bucket], predefined_acl: nil, predefined_default_object_acl: nil, user_project: nil, enable_object_retention: nil, options: {}
+    storage.service.mocked_service = mock
+
+    bucket = storage.create_bucket bucket_name do |b|
+      b.hierarchical_namespace = { enabled: true }
+    end
+
+    mock.verify
+
+    _(bucket).must_be_kind_of Google::Cloud::Storage::Bucket
+    _(bucket.hierarchical_namespace).wont_be_nil
+    _(bucket.hierarchical_namespace.enabled).must_equal true
+  end
+
+  it "creates a bucket with HNS disabled" do
+    mock = Minitest::Mock.new
+    created_bucket = create_bucket_gapi bucket_name
+    created_bucket.hierarchical_namespace = { enabled: false }
```

Comment:
Just a question to make sure my understanding is correct. For a bucket creation without specifying hierarchical namespace configuration,`bucket.hierarchical_namespace` will be nil?

Reply:
Yes, it will be `nil`.